server/infra/database/GrowerAccountRepository.ts

```diff
@@ -341,4 +350,47 @@ export default class GrowerAccountRepository extends BaseRepository<GrowerAccoun
 
     return object.rows;
   }
+
+  async getWalletsCount(filter: GrowerAccountFilter) {
+    const knex = this.session.getDB();
+    const result = await knex
+      .select(
+        knex.raw(`
+        COUNT(DISTINCT(${this.tableName}.id)) AS count
+        FROM ${this.tableName}        
+    `),
+      )
+      .where((builder) => this.filterWhereBuilder(filter, builder))
+      .first();
+
+    return result;
+  }
+
+  async getWalletsByFilter(
+    filter: GrowerAccountFilter,
+    options: FilterOptions,
+  ) {
+    let promise = this.session
+      .getDB()
+      .select(
+        this.session.getDB().raw(`
+        ${this.tableName}.wallet
+        FROM ${this.tableName}
+    `),
+      )
+      .where((builder) => this.filterWhereBuilder(filter, builder))
+      .distinctOn(`${this.tableName}.id`, `${this.tableName}.wallet`)
```

Comment:
Same here as well, wallet is unique, id as well and there are no joins involved, distinct is not needed

Reply:
Done